### DISCUSSION TITLE
Custom template help needed.

### DISCUSSION THREAD

**Discussion by ritchiemarkov** [bootscore/bootscore#425] [Q&A] [answered]
Hello Basti! 
I'm trying to replicate this swiper style:

[https://around.createx.studio/landing-yoga-studio.html](https://around.createx.studio/landing-yoga-studio.html)

After nearly one week my progress : [``](https://www.northwestalps.com/)

Here my template: 

`<?php
/*

 * Post/Page/CPT Hero fade slider template.
 *
 * This template can be overriden by copying this file to your-theme/bs-swiper-main/sc-swiper-hero.php
 *
 * @author 		Bastian Kreiter
 * @package 	bS Swiper
 * @version     5.1.0.0

Posts: 
[bs-swiper-hero-fade type="post" category="cars, boats" order="ASC" orderby="date" posts="6"]

Child-pages: 
[bs-swiper-hero-fade type="page" post_parent="21" order="ASC" orderby="title" posts="6"]

Custom post types:
[bs-swiper-hero-fade type="isotope" tax="isotope_category" terms="dogs, cats" order="DESC" orderby="date" posts="5"]

Single items:
[bs-swiper-hero-fade type="post" id="1, 15"]
[bs-swiper-hero-fade type="page" id="2, 25"]
[bs-swiper-hero-fade type="isotope" id="33, 31"]
*/

// Hero Slider Shortcode
add_shortcode('bs-swiper-landing', 'bootscore_swiper_landing');
function bootscore_swiper_landing($atts) {

  ob_start();
  extract(shortcode_atts(array(
    'type' => 'post',
    'order' => 'date',
    'orderby' => 'date',
    'posts' => -1,
    'category' => '',
    'post_parent'    => '',
    'tax' => '',
    'terms' => '',
    'id' => ''
  ), $atts));

  $options = array(
    'post_type' => $type,
    'order' => $order,
    'orderby' => $orderby,
    'posts_per_page' => $posts,
    'category_name' => $category,
    'post_parent' => $post_parent,
  );

  $tax = trim($tax);
  $terms = trim($terms);
  if ($tax != '' && $terms != '') {
    $terms = explode(',', $terms);
    $terms = array_map('trim', $terms);
    $terms = array_filter($terms);
    $terms = array_unique($terms);
    unset($options['category_name']);
    $options['tax_query'] = array(array(
      'taxonomy' => $tax,
      'field'    => 'name',
      'terms'    => $terms,
    ));
  }

  if ($id != '') {
    $ids = explode(',', $id);
    $ids = array_map('intval', $ids);
    $ids = array_filter($ids);
    $ids = array_unique($ids);
    $options['post__in'] = $ids;
  }

  $query = new WP_Query($options);
  if ($query->have_posts()) { ?>

<section class="ignore-dark-mode position-relative min-vh-100 height-100 width-100">
  
  <div class="heroes swiper swiper-scale-effect position-absolute top-0 start-0 w-100 h-100" data-swiper-options='{
      "effect": "fade",
      "speed": 800,
      "autoplay": {
        "delay": 7000,
        "disableOnInteraction": false
      },
      "pagination": {
        "el": ".swiper-pagination",
        "clickable": true
      }
    }'>

    <div class="swiper-wrapper">
      <?php while ($query->have_posts()) : $query->the_post(); ?>
        <div class="swiper-slide">
          <div class="swiper-slide-cover bg-position-top-center" style="background-image: url('<?php echo get_the_post_thumbnail_url(get_the_ID(), 'full'); ?>');"></div>
          <div class="position-absolute top-0 end-0 bottom-0 start-0">
            <div class="container h-100 d-flex flex-column">
              <div class="mt-auto text-white mb-5 text-center">
                <?php bootscore_category_badge(); ?>
                <h2 class="blog-post-title">
                  <a class="text-white" href="<?php the_permalink(); ?>">
                    <?php the_title(); ?>
                  </a>
                </h2>
                <div class="card-text">
                  <?php the_excerpt(); ?>
                </div>
                <a class="read-more btn btn-light" href="<?php the_permalink(); ?>"><?php _e('Read more »', 'bootscore'); ?></a>
                <?php bootscore_tags(); ?>
              </div>
            </div>
          </div>
        </div>
      <?php endwhile;
      wp_reset_postdata(); ?>
    </div> 

    
    <div class="swiper-pagination"></div>
    
    <div class="swiper-button-next d-none d-lg-block"></div>
    <div class="swiper-button-prev d-none d-lg-block"></div>

  </div>
  
</section>
<?php $myvariable = ob_get_clean();
return $myvariable;
}
}
`

I know you might be busy, but i bet you solve this quickly (after 1 week im pretty surrended).
If its working, i can PR this template since its pretty usefull to have it expecially on landing pages.
Thank you so much!

**Comment by crftwrk** [comment, marked as answer]
Hi Richard,

I'm not in front of my machine until middle of next week. So, I cannot check your code. But this seems not to be a big deal.

- Use `height-100` class for full height swiper https://bootscore.me/documentation/width-height-classes/#Using_height-_classes
- Set featured image to `objet-fit: cover;`
- Add some CSS to create Ken Burns effect to featured images https://stackoverflow.com/questions/53898585/how-can-i-add-the-ken-burns-effect-to-swiper
- Add some CSS to style the bullets
- Remove the arrow pagination in the HTML (last 2 lines in your code)
- The heading seems to be outside the `swiper-slide`. So, delete title, excerpt, read more etc and hardcode headline, paragraph and video button inside the `swiper-wrapper` but outside the `swiper-slide`
- Add some CSS to position this where you want.

But as far as I know, createx.studio uses swiper everywhere. Why not simply grab the HTML and CSS in your browser console and adapt it to your template?

**Comment by ritchiemarkov** [comment]
Thank you as always for your exhaustive answer. I actually did! I copied
the whole code but they were using background for image. So I had to
understand how to implement it into php. The most annoying thing that they
uses several sass files with their own variables. So I had also to
understand where and why.

I understand that it's not so complicated for experienced dev but for me
it's black magic 😭. So I didn't understand if width 100 height 100 is OK
for usage. I hope I will figure out why  this little piece of next slide is
coming out, and eventually will share it.

I will mark this one as answered. But if you have any spare time, just take
a look please. Thank you once again for your support bro.

Il Mar 7 Mar 2023, 17:48 Basti ***@***.***> ha scritto:

**Reply by crftwrk** [reply]
Yes, we can check this by the end of the next week, pls remind me ;-)
